### ISSUE TITLE
PRVHASH 2.22

### ISSUE THREAD

**Issue by avaneev** [rurban/smhasher#159]
Hello! I've updated PRVHASH to v2.22 - now the raw speed of `prvhash42` is a lot faster, it is also still fast when used as 32-bit table hash. Other hash lengths are irrelevant for tables anyway. However, both `prvhash42` and `prvhash42s` now require scratch pad bytes in Hash pointer: imposed after extensive research; this way the "space" of random number generator is exponentially larger. This isn't important for short strings, but when hashing gigabytes of data this is important.

32-bit hashes do not require scratch pad, but other hash lengths should now be acquired this way:
```
	const int hl = 8;
	uint8_t Tmp[ hl * 2 ];
	prvhash42( (const uint8_t*) key, len, Tmp, hl, seed, 0 );
	memcpy( out, Tmp, hl );
```

https://github.com/avaneev/prvhash

**Comment by avaneev** [comment]
Both functions should be completely retested as they now have different statistical properties. MomentChi admittedly sometimes goes a little bit above 4 for bit 0 or 1, but I think it's not a problem - it shows a kind of natural assymetry, MomentChi is usually not large for both bit 0 and 1 simultaneously.

**Comment by rurban** [comment]
The problem is that prvhash42_128 is now invalid, and prvhash42_64 also for 32bit CPU's.
We only allocate 128 bit hashes for 128.

**Comment by rurban** [comment]
Merged as 2.22a variant
